src/presentations/indexable-post-type-presentation.php

```diff
@@ -337,4 +291,84 @@ public function generate_twitter_creator() {
 
 		return '';
 	}
+
+	/**
+	 * Creates a paginated canonical for the current page.
+	 *
+	 * @param boolean $add_pagination_base Whether the pagination base should be added.
+	 *
+	 * @return string The canonical.
+	 */
+	protected function build_paginated_canonical( $add_pagination_base ) {
+		if ( $this->model->canonical ) {
+			return $this->model->canonical;
+		}
+
+		$canonical = $this->model->permalink;
+
+		// Fix paginated pages canonical, but only if the page is truly paginated.
+		$current_page = $this->pagination->get_current_post_page_number();
+		if ( $current_page > 1 ) {
+			$number_of_pages = $this->model->number_of_pages;
+			if ( $number_of_pages && $current_page <= $number_of_pages ) {
+				$canonical = $this->pagination->get_paginated_url( $canonical, $current_page, $add_pagination_base );
+			}
+		}
+
+		return $this->url->ensure_absolute_url( $canonical );
+	}
+
+	/**
+	 * Creates a rel prev link for the current page.
+	 *
+	 * @param boolean $add_pagination_base Whether the pagination base should be added.
+	 *
+	 * @return string The rel prev link.
+	 */
+	protected function build_rel_prev( $add_pagination_base ) {
+		if ( $this->model->number_of_pages === null ) {
+			return '';
+		}
+
+		if ( $this->pagination->is_rel_adjacent_disabled() ) {
+			return '';
+		}
+
+		$current_page = \max( 1, $this->pagination->get_current_post_page_number() );
+		// Check if there is a previous page.
+		if ( $current_page < 2 ) {
+			return '';
+		}
+		// Check if the previous page is the first page.
+		if ( $current_page === 2 ) {
+			return $this->model->permalink;
+		}
+
+		return $this->pagination->get_paginated_url( $this->model->permalink, ( $current_page - 1 ), $add_pagination_base );
+	}
+
+	/**
+	 * Creates a rel next link for the current page.
+	 *
+	 * @param boolean $add_pagination_base Whether the pagination base should be added.
+	 *
+	 * @return string The rel next link.
+	 */
+	protected function build_rel_next( $add_pagination_base ) {
```

Comment:
Instead of adding new build method I would consolidate the piece that causes this addition. In this case it is:
```php
return $this->pagination->get_paginated_url( $this->model->permalink, ( $current_page + 1 ), $add_pagination_base );
```
In this case it is `$add_pagination_base` that might be differ. I suggest to make a wrapper method for this:
```
protected function get_paginated_url( $url, $page ) {
    return $this->pagination->get_paginated_url( $url, $page, true );
```

This allows you to have the generate method having the same content as it is now, but the only difference is the 
call of `return $this->pagination->get_paginated_url` that becomes: `$this->get_paginated_url( $this->model->permalink, ( $current_page + 1 ) )`.

By doing this you've just added one single method and you can keep the rest as it is now.